pandas-stubs/core/series.pyi

```diff
@@ -900,18 +906,27 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
     @overload
     def dropna(
         self,
-        axis: SeriesAxisType = ...,
-        how: Literal["any", "all"] | None = ...,
         *,
+        axis: SeriesAxisType = ...,
         inplace: Literal[True],
+        how: Literal["any", "all"] | None = ...,
     ) -> None: ...
     @overload
     def dropna(
         self,
+        *,
         axis: SeriesAxisType = ...,
-        inplace: _bool = ...,
+        inplace: Literal[False] = ...,
         how: Literal["any", "all"] | None = ...,
     ) -> Series[S1]: ...
+    @overload
+    def dropna(
+        self,
+        *,
+        axis: SeriesAxisType = ...,
+        inplace: _bool = ...,
+        how: Literal["any", "all"] | None = ...,
+    ) -> Series[S1] | None: ...
```

Comment:
Not clear why we need to add this last overload

Reply:
I think you can be more specific since `inplace: Literal[True]` is the only way to get `None`.  It is already positional only, so you can make it Literal[True] and remove the default argument.

Reply:
Not sure whether that is used in practise but needed for cases like:

```py
def test(series: pd.Series, inplace: bool):
    series.reset_index(inplace=inplace)  # None or pd.Series
    # without the overload, mypy would complain
```

Reply:
I think you can still make it after the specific overload for `Literal[True]->None` which will be slightly more specific when using `inplace=True`

Reply:
The first overload already covers the `True` case.

Reply:
I'm now fine with this change.  It's consistent with what we did with other methods in `series.pyi` where the last declaration covers the `inplace: _bool = ...` case giving a `Union` as the result.